doc/releases/changelog-dev.md

```diff
@@ -112,9 +112,39 @@
   ```
 
   ``` pycon
-    >>> workflow(np.array([2.0, 1.0]))
-    array([[-1.32116540e-07,  1.33781874e-07],
-           [-4.20735506e-01,  4.20735506e-01]])
+  >>> workflow(np.array([2.0, 1.0]))
+  array([[-1.32116540e-07,  1.33781874e-07],
+          [-4.20735506e-01,  4.20735506e-01]])
+  ```
+
+* `qml.for_loop` and `qml.while_loop` compiler-specific decorators are added.
+  `qml.cond` can be used with the `qml.qjit` decorator.
+  [(#4698)](https://github.com/PennyLaneAI/pennylane/pull/4698)
+
+  ``` python
+  dev = qml.device("lightning.qubit", wires=1)
+
+  @qml.qjit
+  @qml.qnode(dev)
+  def circuit(n: int, x: float):
+
+      @qml.for_loop(0, n, 1)
+      def loop_rx(i, x):
+          # perform some work and update (some of) the arguments
+          qml.RX(x, wires=0)
+
+          # update the value of x for the next iteration
+          return jnp.sin(x)
+
+      # apply the for loop
+      final_x = loop_rx(x)
+
+      return qml.expval(qml.PauliZ(0)), final_x
```

Comment:
I know we're expecting some differences in what can be done with pennylane+catalyst (compared to basic pennylane), but I'm a little surprised to have a QNode returning something that's not a measurement process.

firstly, do we want to support this? second, even if we do (which we already do so why not I guess), do we want to advertise it? kinda breaks some long-standing expectations (hehe) of pennylane

Reply:
Excellent point! There's no such limitation with returning values in Catalyst. It's expected and in fact one of the unique features of Catalyst. imo we should not advertise it as we don't discuss such differences between Catalyst and PL for this soft-integration and it's better to update the example in the changelog. cc: @josh146 @dime10

Reply:
if you do choose to remove 'em, don't forget to remove them from the doc examples for `while_loop` and `for_loop`. anyway, sounds good